ddl/partition.go

```diff
@@ -1911,27 +1911,152 @@ func onTruncateTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (int64, e
 		return ver, errors.Trace(dbterror.ErrPartitionMgmtOnNonpartitioned)
 	}
 
-	newPartitions := make([]model.PartitionDefinition, 0, len(oldIDs))
-	for _, oldID := range oldIDs {
-		for i := 0; i < len(pi.Definitions); i++ {
-			def := &pi.Definitions[i]
-			if def.ID == oldID {
-				pid, err1 := t.GenGlobalID()
-				if err1 != nil {
-					return ver, errors.Trace(err1)
+	switch job.SchemaState {
+	case model.StatePublic:
```

Comment:
Should we simplify this if there are no global index and only have a single state change where the partitions will have the new ids? Or do we still need to go through the different stages?

Reply:
`onDropTablePartition` also use Public->DeleteOnly->DeleteReorg 3 stages. Here I just stay the same. If you think it is necessary, I will improve the design.